### ISSUE TITLE
checkhealth and subreports - all generate E28: No such highlight group markdownError

### ISSUE THREAD

**Issue by EdmundsEcho** [neovim/neovim#17008]
### Neovim version (nvim -v)

NVIM v0.6.1 Build type: Release LuaJIT 2.1.0-beta3 Compiled by brew@HMBRW-A-001-M1-004.local

### Vim (not Nvim) behaves the same?

I believe checkhealth is nvim specific

### Operating system/version

macOS 12.1

### Terminal name/version

alacritty

### $TERM environment variable

screen-256color

### Installation

homebrew

### How to reproduce the issue

:checkhealth nvim
:checkhealth nvim-treesitter
:checkhealth vim.lsp

...etc

Notes: 

* `:checkhealth` on its own fails to report anything other than the error.
*  I get the same behavior in versions 6 & 7 of neovim

### Expected behavior

Report the result without generating the error.

Per the note: running `:checkhealth` should generate the report.

### Actual behavior

When I run the specific tests, I get both the error and the report.  When I run the group of reports, I just get the error.

```
Error detected while processing /opt/homebrew/Cellar/neovim/0.6.1/share/nvim/runtime/syntax/checkhealth.vim:
line   15:
E28: No such highlight group name: markdownError
Error detected while processing function health#check:
line   33:
E684: list index out of range: 0
```

**Comment by clason** [comment]
That's already fixed in https://github.com/neovim/neovim/pull/16930.

**Comment by EdmundsEcho** [comment]
I see what you are saying.  In the situation I described, it turns out the route cause was a plugin that does not return anything in the check health routine.  I was able to fix it by adding a guard in the checkhealth script that throws an error in such an event.  This is the behavior already in place when neovim doesn't get what it expects from a plugin checkhealth routine.  The extra guard is in the existing try/catch block.  I posted the suggested change in the PR #17012.
